central/securedclustercertgen/certificates.go

```diff
@@ -13,33 +13,64 @@ import (
 // secretDataMap represents data stored as part of a secret.
 type secretDataMap = map[string][]byte
 
-var v2ServiceTypes = set.NewFrozenSet[storage.ServiceType](storage.ServiceType_SCANNER_SERVICE, storage.ServiceType_SCANNER_DB_SERVICE)
-var v4ServiceTypes = set.NewFrozenSet[storage.ServiceType](storage.ServiceType_SCANNER_V4_INDEXER_SERVICE, storage.ServiceType_SCANNER_V4_DB_SERVICE)
+var scannerV2ServiceTypes = set.NewFrozenSet[storage.ServiceType](storage.ServiceType_SCANNER_SERVICE, storage.ServiceType_SCANNER_DB_SERVICE)
+var scannerV4ServiceTypes = set.NewFrozenSet[storage.ServiceType](storage.ServiceType_SCANNER_V4_INDEXER_SERVICE, storage.ServiceType_SCANNER_V4_DB_SERVICE)
+var localScannerServiceTypes = func() set.FrozenSet[storage.ServiceType] {
+	types := scannerV2ServiceTypes.Unfreeze()
+	types = types.Union(scannerV4ServiceTypes.Unfreeze())
+	return types.Freeze()
+}()
+
+var securedClusterServiceTypes = set.NewFrozenSet[storage.ServiceType](
+	storage.ServiceType_SENSOR_SERVICE,
+	storage.ServiceType_COLLECTOR_SERVICE,
+	storage.ServiceType_ADMISSION_CONTROL_SERVICE)
+
 var allSupportedServiceTypes = func() set.FrozenSet[storage.ServiceType] {
-	types := v2ServiceTypes.Unfreeze()
-	types = types.Union(v4ServiceTypes.Unfreeze())
+	types := securedClusterServiceTypes.Unfreeze()
+	types = types.Union(localScannerServiceTypes.Unfreeze())
 	return types.Freeze()
 }()
 
+type certIssuerImpl struct {
+	allSupportedServiceTypes set.FrozenSet[storage.ServiceType]
```

Comment:
TBH this whole check [here](https://github.com/stackrox/stackrox/pull/12740/files#diff-1a886f7e4f8d2a5cec66f1de6f741c310779304249c2aa6f53901ba498e40149R112) to me seems redundant and I would remove it. (see my other reply [here](https://github.com/stackrox/stackrox/pull/12740#discussion_r1775024664))

Basically we define sets of certificates, and then in `generateServiceCertMap` we check again if the cert to be generated is of an accepted type, even though it's an unexported function. IMO the check in the unit tests should be enough here.

Reply:
> By definition the service types associated with a certIssuer would then be the set of services for which the certIssuer can issue certificates (and hence this would "all supported" part would be implicit in it).

I agree, and then I can also remove the parameter `serviceTypes` of `issueCertificates`. It will alter slightly the check [here](https://github.com/stackrox/stackrox/pull/12740/files#diff-1a886f7e4f8d2a5cec66f1de6f741c310779304249c2aa6f53901ba498e40149R112) but it won't have any real consequences (I actually find that check redundant and I'd remove it).